core/src/main/scala/cats/data/ReaderWriterStateT.scala

```diff
@@ -42,18 +42,6 @@ final class ReaderWriterStateT[F[_], E, S, L, A](val runF: F[(E, S) => F[(L, S,
     transform { (l, s, a) => (f(l), s, a) }
 
   /**
-   * Combine this computation with `rwsb` using `fn`. The state will be be threaded
-   * through the computations and the log values will be combined.
-   */
-  def map2[B, Z](rwsb: ReaderWriterStateT[F, E, S, L, B])(fn: (A, B) => Z)(
-    implicit F: FlatMap[F], L: Semigroup[L]): ReaderWriterStateT[F, E, S, L, Z] =
-    flatMap { a =>
-      rwsb.map { b =>
-        fn(a, b)
-      }
-    }
```

Comment:
You can get the same from `Apply#map2`

The version from `Apply#map2` would look like : 

```scala
map(flatMap(fa)(a => map(fb)(b => (a, b)))) { case (a, b) => f(a, b) }
``` 

If we want the simpler implementation used here in `RWST`, we could override `map2` in `FlatMap`.